File: Lab-FInal/3.c
#include<stdio.h>
#include<stdlib.h>
#include<signal.h>
#include<pthread.h>
#include<unistd.h>

void *thread_task() 
{
    while (1) 
    {
        printf("Thread task is running...\n");
        sleep(1);
    }
    return NULL;
}

void sigint_handler() 
{
    printf("\nSIGINT signal caught. Terminating program.\n");
    exit(0);
}

int main() 
{
    signal(SIGINT,sigint_handler);
    pthread_t tid;
    if (pthread_create(&tid,NULL,thread_task,NULL)!=0) 
    {
        perror("pthread_create");
        exit(1);
    }
    while(1) 
    {
        printf("Main program is running...\n");
        sleep(1);
    }
    return 0;
}

File: Lab-FInal/3.py
import signal
import threading
import time
import sys

def thread_task():
    while True:
        print("Thread task is running...")
        time.sleep(1)

def sigint_handler(signum, frame):
    print("\nSIGINT signal caught. Terminating program.")
    sys.exit(0)

def main():
    signal.signal(signal.SIGINT,sigint_handler)

    thread=threading.Thread(target=thread_task)
    #thread.daemon=True  
    thread.start()

    while True:
        print("Main program is running...")
        time.sleep(1)

if __name__ == "__main__":
    main()
